### PR TITLE
Github action to check if quiche new version/tag exists. if current project not up2date we create an issue

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

File: .github/workflows/check-quiche-release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        run: |
+          set -x
+          LAST_TAG=$(gh release view -R cloudflare/quiche --json tagName --jq '.tagName')
+          echo $LAST_TAG
+          CURRENT_QUICHE_VERSION=$(grep -A1 "## Current quiche version" README.md|grep -v "## Current quiche version")
+          echo $CURRENT_QUICHE_VERSION
+          if [ $LAST_TAG != $CURRENT_QUICHE_VERSION ] ; then
+            if gh issue list --search "Quiche $LAST_TAG" --json title | jq -e ". == []" ; then
+              echo "current quiche version $CURRENT_QUICHE_VERSION is not the same as last tag $LAST_TAG create issue"
+              gh issue create -t "Quiche $LAST_TAG" -b "Update to Quiche $LAST_TAG"
```
Comment (joakime) [inline .github/workflows/check-quiche-release.yml:29]:
```suggestion
            ISSUE_TITLE="Bump Quiche $LAST_TAG"
            if gh issue list --search "$ISSUE_TITLE" --json title | jq -e ". == []" ; then
              echo "current quiche version $CURRENT_QUICHE_VERSION is not the same as last tag $LAST_TAG create issue"
              gh issue create -t "$ISSUE_TITLE" -b "Bump Quiche version from $CURRENT_QUICHE_VERSION to $LAST_TAG\n * https://github.com/cloudflare/quiche/releases/tag/$LAST_TAG"
```

**Review by joakime — COMMENTED** [review]
Small improvements

File: .github/workflows/check-quiche-release.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          
+      - name: Get Quiche Release
+        shell: bash
+        env:
+          GH_TOKEN: ${{ github.token }}
```
Comment (joakime) [inline .github/workflows/check-quiche-release.yml:19]: Isn't this ... ?

```suggestion
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}        
```

**Review by joakime — CHANGES_REQUESTED** [review]
Lets experiment somewhere else (esp around the token)

Fork this repo into your personal org level.
Revert main back to the pre-0.20.0 state.
Add this new action, and lets see it work.

**Comment by olamy** [comment]
Already tested here https://github.com/olamy/jetty-quiche-native
Check issues and modified readme

File: .github/workflows/check-quiche-release.yml (hunk 2 above)
Comment (olamy) [reply .github/workflows/check-quiche-release.yml:19]: that's perfectly fine and equivalent https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

**Comment by olamy** [comment]
closed for #81

File: .github/workflows/check-quiche-release.yml (hunk 1 above)
Comment (olamy) [reply .github/workflows/check-quiche-release.yml:29]: integrated in other PR

**Pull request closed without merging**